src/compute-client/src/plan/join/linear_join.rs

```diff
@@ -191,10 +191,14 @@ impl LinearJoinPlan {
                 mfp.permute(permutation.clone(), key.len() + thinning.len());
                 let mfp = mfp.into_plan().unwrap().into_nontemporal().unwrap();
                 (
-                    Some(JoinClosure {
-                        ready_equivalences: vec![],
-                        before: mfp,
-                    }),
+                    if mfp.is_identity() {
+                        None
+                    } else {
+                        Some(JoinClosure {
+                            ready_equivalences: vec![],
+                            before: mfp,
+                        })
+                    },
```

Comment:
Can we do the identity check before calling `mfp.into_plan()`? I'm not sure how expensive that operation is, but it doesn't look cheap at least.

Reply:
`mfp.into_plan()` runs `MapFilterProject::optimize` (among other things), and I can imagine situations where `is_identity()` wouldn't realize that it's an identity from looking at the unoptimized MFP, but realizes it from the optimized MFP. For example, `remove_undemanded` might remove some superfluous expressions.